src/cpu/testers/dr_trace_player/trace_player.cc

```diff
@@ -167,24 +165,59 @@ DRTracePlayer::executeMemInst(DRTraceReader::TraceRef &mem_ref)
 bool
 DRTracePlayer::trySendMemRef(DRTraceReader::TraceRef &mem_ref)
 {
-    PacketPtr pkt = getPacket(mem_ref);
+    PacketPtr pkt, split_pkt;
+
+    std::tie(pkt, split_pkt) = getPacket(mem_ref);
+
+    if (!retrySplitPkt) {
+        // we should not be sending the first pkt again if
+        // we are retrying only on the second part of the
+        // previously stalled request
+        DPRINTF(DRTrace, "Trying to send %s\n", pkt->print());
+
+        if (!port.sendTimingReq(pkt)) {
+            DPRINTF(DRTrace, "Failed to send pkt\n");
+            if (stats.memStallStart == 0) {
+                stats.memStallStart = curTick();
+            }
+            numOutstandingMemReqs++;
+            stats.outstandingMemReqs.sample(numOutstandingMemReqs);
+
+            delete pkt;
+            if (split_pkt == nullptr) {
+            // if this is not a split pkt req,
+            // we can return back here
+            return true;
+            }
+        } else {
+            stats.latencyTracker[pkt] = curTick();
+            return false;
+        }
+    }
 
-    DPRINTF(DRTrace, "Trying to send %s\n", pkt->print());
+    DPRINTF(DRTrace, "Trying to send split %s\n", split_pkt->print());
```

Comment:
We should only send the second packet when the first packet successfully sends.

Reply:
This has not been addressed.

Reply:
I think I have not fully understood this concern/comment. This DPRINTF statement will be executed only if we successfully send out the first packet.

Reply:
Sorry for the confusion. This wasn't a comment on the DPRINTF. It's been addressed.